### PR TITLE
Add support to Editor Drag&Drop

### PR DESCRIPTION
**Pull request by zhuxingwei**
This PR aims to add Drag&Drop support for UIWigetsEditorWindow, which enables features like:
(1) user can drag a Unity GameObject into a UIWigetsEditorWindow and drop it inside this window;
(2) user can drag any file from the OS (e.g., FileOpenDialog or Finder (on Mac)) into a UIWigetsEditorWindow and drop it inside this window;

**This PR also modifies UIWidgets libraries**.

How to Use:
(1) You need to use `UIWidgetsEditorBinding` to create the App in the `main` method of your UIWidgetsEditorPanel like:
```
protected override void main()
{
      editor_ui_.runEditorApp(new YourApp());
}
```
(2) The functionality is wrapped inside a widget: `UnityObjectDetector`, in which there are four callbacks that can be triggered when:
(A) dragging UnityObject or file to enter the widget; (B) dragging UnityObject or file to exit the widget; (C) dragging UnityObject or file and release it inside the widget; (D) dragging UnityObject or file and moving inside the widget

How to Test:
You can find a test sample in the Editor main menu under `UIWidgets/EditorSample/UnityObjectDragging`

### CODE REVIEW

**Comment by guanghuispark** [comment]
Now that we have made this change, do we need to update the dll file?

**Comment by zhuxingwei** [comment]
> Now that we have made this change, do we need to update the dll file?

yes we do. I have updated the Mac library (Debug version) so that you can test the functionality on your Mac. I will update other libraries once this PR is approved

File: com.unity.uiwidgets/Editor/gesture/editor_mouse_tracking.cs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        ) {
+            D.assert(router != null);
+            D.assert(annotationFinder != null);
+            router.addGlobalRoute(route: _handleEvent);
+            this.annotationFinder = annotationFinder;
+        }
+
+        public void _handleEvent(PointerEvent evt) {
+
+            //Debug.Log("handle editor mouse tracker event = " + evt.position);
```
Comment (SiyaoHuang) [inline com.unity.uiwidgets/Editor/gesture/editor_mouse_tracking.cs:92]: remove test code.

**Review by SiyaoHuang — APPROVED** [review]

File: com.unity.uiwidgets/Editor/gesture/editor_mouse_tracking.cs (hunk 1 above)
Comment (zhuxingwei) [reply com.unity.uiwidgets/Editor/gesture/editor_mouse_tracking.cs]: updated

**Comment by zhuxingwei** [comment]
> > Now that we have made this change, do we need to update the dll file?
> 
> yes we do. I have updated the Mac library (Debug version) so that you can test the functionality on your Mac. I will update other libraries once this PR is approved

all libraries are updated and tested

**Review by guanghuispark — APPROVED** [review]

**Pull request merged**